File: 0392-is-subsequence/0392-is-subsequence.java
class Solution {
    public boolean isSubsequence(String s, String t) {
        
        if(s.length() == 0) return true;
  
        int p1 = 0; 
        for(int i=0; i<t.length(); i++){
            if(s.charAt(p1) == t.charAt(i)){
                p1++;
                if(p1 == s.length()) return true;
            }
        }
        
        
        return false;
        
    }
}

File: 0392-is-subsequence/0392-is-subsequence.js
/**
 * @param {string} s
 * @param {string} t
 * @return {boolean}
 */
var isSubsequence = function(s, t) {

    var cnt = 0;
    var j = 0;
    for(var i=0; i<t.length; i++){
        if(t[i] === s[j]) j++;
    }

   return j === s.length ? true : false;
};